src/converter/internal/compat/backendapi.cpp

```diff
@@ -54,7 +54,7 @@ static const std::string SEGMENTS_POSITIONS_WRITER_NAME = "sposXML";
 static const std::string MEASURES_POSITIONS_WRITER_NAME = "mposXML";
 static const std::string PDF_WRITER_NAME = "pdf";
 static const std::string MIDI_WRITER_NAME = "midi";
-static const std::string MUSICXML_WRITER_NAME = "mxml";
+static const std::string MUSICXML_WRITER_NAME = "mxl";
```

Comment:
I suspect that this is supposed to be `mxml`. But keep it in the PR for now.

It looks like the string `mxml` used in the website API (it appears in the 3.x code [here](https://github.com/musescore/MuseScore/blob/2513676e512d29d554cb6c4d37d3efaf53ea2c5b/mscore/file.cpp#L3569)), and this probably caused the original confusion that saw [`.mxml` added as an export format](https://github.com/musescore/MuseScore/pull/8344/files#diff-b3721830027f45451d2b81cff70b29584ca58ec71e95b07cd2effe26d8f33f5bR74). I could be wrong though so let's wait for confirmation.

Reply:
It also is uesd as the extension apparently, that at least is my understanding.

Reply:
Actually delving deeper into the code I'm now pretty sure that this `mxml` is **also** used as the extension, not **just** as a key in the metadata.json

Reply:
So we may also want to change `midi` to `mid`, to keep the downloadable formats the same as with MU3 (and a 3 char extension maximises compatibility)

Reply:
`mxml` is fixed in #13175.

I kept `midi` as it is. It seems that `.midi` is widely recognised as a MIDI file even if `.mid` is the normal extension. Both MU3 and MU4 allow the creation of MIDI files with `.mid`, `.midi` and even `.kar` as the extension. This is achieved in the GUI by manually changing the file extension in the operating system's Export dialog that appears after our Export dialog.

<img width="598" alt="image" src="https://user-images.githubusercontent.com/11011881/188511097-2487a779-7ff6-4dc8-8258-ea7a1f86cbe6.png">

We don't want dedicated options for `.midi` and `.kar` because:

1. `.mid` is preferred over `.midi`.
2. `.kar` is not quite the same as a MIDI file. As I understand, the Karaoke format is a subset of MIDI files where only the top staff has lyrics. To properly support this we would need to:
    1. Actually export lyrics to MIDI files (currently we don't export any lyrics to MIDI file).
    2. Prevent lyrics being exported to all except the top staff when `.kar` is selected.

Reply:
That's not the point: Up till now MuseScore.com offeres a MIDI download as ".mid". and also the MuseScore program exports as ".mid" only.
Now, at least on musescore.com, that apparently changes to ".midi", and that does seem inconsistent.

Reply:
I think you might have misunderstood what the `--score-media` command does. It doesn't write any files to disk. It simply sends this JSON data to `stdout`:
```JSON
{
"pngs":  ["BASE64DATA"],
"svgs":  ["BASE64DATA"],
"sposXML": "BASE64DATA",
"mposXML": "BASE64DATA",
"pdf": "BASE64DATA",
"midi": "BASE64DATA",
"mxml": "BASE64DATA",
"metadata": {"composer":"Composer","duration":4,"fileVersion":302,"hasHarmonies":"false","hasLyrics":"false","keysig":0,"lyrics":"","measures":2,"mscoreVersion":"3.6.2","pageFormat":{"height":297,"twosided":"true","width":210},"pages":1,"parts":[{"harmonyCount":0,"hasDrumStaff":"false","hasPitchedStaff":"true","hasTabStaff":"false","instrumentId":"keyboard.piano","isVisible":"true","lyricCount":0,"name":"Piano","program":0}],"poet":"","previousSource":"","subtitle":"","tempo":0,"tempoText":"","textFramesData":{"composers":["Composer"],"poets":[],"subtitles":[],"titles":["Title"]},"timesig":"4/4","title":"Title"}
}
```
So the output contains the *contents* of a MIDI file (as [Base64](https://en.wikipedia.org/wiki/Base64#Base64_table_from_RFC_4648) data); it doesn't specify a *name* for the MIDI file, therefore the website is free to call the file whatever it likes.

The output (including the names of the `"midi"` and `"mxml"` object keys) has not changed since MuseScore 3, yet the website is already offering those files with the correct `.mid` and `.mxl` extensions. PNG and SVG images are being used correctly on the website despite those key names in this file ending in the letter "s".

Reply:
As far as I can tell it does not only that, but **also** does the corresponding export

Reply:
Check `PNG_WRITER_NAME` and `SVG_WRITER_NAME? :  it is "png" resp. "svg", singular, but in your `score-media` output it is plural. Check `SEGMENTS_POSITIONS_WRITER_NAME` and `MEASURES_POSITIONS_WRITER_NAME` and compare with 
```
        writers->reg({ "sposXML" }, std::make_shared<PositionsWriter>(PositionsWriter::ElementType::SEGMENT));
        writers->reg({ "mposXML" }, std::make_shared<PositionsWriter>(PositionsWriter::ElementType::MEASURE));
```
and how that differes from MuseScore 3

Reply:
It has to *generate* a MIDI file in order to embed it within the JSON as Base64 data. This involves running the MIDI export code, but the file is "exported" into memory, not saved to the hard drive.

Reply:
Ah, I see! So the backend still produces ".mid" files for download?

Reply:
Still my change won't harm here, or would it?

Reply:
> So the backend still produces ".mid" files for download?

Yes.

> Still my change won't harm here, or would it?

It does no harm because you kept the JSON key as `"midi"`, but it's not really necessary or related to this PR.

Reply:
OK, moved to another PR